.claude/skills/run-synthetic-factory/driver.py

```
@@ -116,6 +116,34 @@ def marker_visible_jsonl_paths(run_dir):
+        temp, snap = snapshot_to_temp(src, prefix)
+        try:
+            visible_after = marker_visible_jsonl_paths(src)
+        except BaseException:
+            temp.cleanup()
+            raise
+        if visible_before == visible_after:
+            return temp, snap, visible_before
+        temp.cleanup()
+    raise TransactionError(
+        f"run changed transaction visibility repeatedly while snapshotting: {src}"
+    )
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)